core/scheduler/service/replay_validator.go

```diff
@@ -33,26 +33,28 @@ func (v Validator) Validate(ctx context.Context, replayRequest *scheduler.Replay
 	return v.validateConflictedRun(ctx, replayRequest, jobCron)
 }
 
-func (v Validator) validateDateRange(ctx context.Context, replayRequest *scheduler.Replay) error {
+func (v Validator) validateDateRange(ctx context.Context, replayRequest *scheduler.Replay, jobCron *cron.ScheduleSpec) error {
 	jobSpec, err := v.jobRepo.GetJobDetails(ctx, replayRequest.Tenant().ProjectName(), replayRequest.JobName())
 	if err != nil {
 		return err
 	}
 	replayStartDate := replayRequest.Config().StartTime.UTC()
 	replayEndDate := replayRequest.Config().EndTime.UTC()
-	jobStartDate := jobSpec.Schedule.StartDate.UTC()
-	jobEndDate := time.Now().UTC()
+	jobExecutionStartDate := jobSpec.Schedule.StartDate.UTC()
+	schedulingStartDate := jobCron.Next(jobExecutionStartDate)
+
 	// time bound for end date
-	if jobSpec.Schedule.EndDate != nil && jobSpec.Schedule.EndDate.UTC().Before(jobEndDate) {
-		jobEndDate = jobSpec.Schedule.EndDate.UTC()
+	jobEndDate := jobSpec.Schedule.EndDate.UTC()
```

Comment:
`jobSpec.Schedule.EndDate` is nullable, so might cause panic

Reply:
sure